### ISSUE TITLE
Incorrect placement inside of fixed container with transforms

### ISSUE THREAD

**Issue by hgascoigne** [FezVrasta/popper.js#56]
I have an application with a drawer with `position: fixed` and is pushed in and out of the interface using transforms. I've found that placing a Popper within that drawer causes the position of that Popper to be way outside of the viewport. It seems that this is caused by the transform on the parent drawer because the transform origin is no longer relative to `document.body` but the drawer element.

Here's a [CodePen](http://codepen.io/hgascoigne/pen/YWNeBw) illustrating the issue. Uncomment the `transform: translateX(0)` line in the CSS to see what I mean. I'm going to try and work up a solution on my end, but since I don't know your code very well I wanted to put it up here so you might find a better way than me to fix it.

**Comment by FezVrasta** [comment]
What if you set the parent as relative+

**Comment by FezVrasta** [comment]
Will look into it once I get a good internet connection. Expect a fix in not less than a week sorry.

**Comment by hgascoigne** [comment]
Simply setting the parent as relative does not work because elements with a fixed position have an offsetParent of null, so the `getOffsetParent()` function just returns `body` which will cause the offsets to be based off the top left of the viewport, when it should be the top left of its offset parent.

I was able to solve the problem by changing this:

```
//
// Get reference element position
//
var referenceOffsets = getOffsetRectRelativeToCustomParent(reference, getOffsetParent(popper), isParentFixed);
```

to this:

```
//
// Get reference element position
//
var offsetParent = isParentFixed ? getOffsetParent(reference) : getOffsetParent(popper)
var referenceOffsets = getOffsetRectRelativeToCustomParent(reference, offsetParent, isParentFixed);
```

and removing this from `getOffsetRectRelativeToCustomParent()`:

```
if (fixed) {
  var scrollParent = getScrollParent(parent);
  parentRect.top += scrollParent.scrollTop;
  parentRect.bottom += scrollParent.scrollTop;
  parentRect.left += scrollParent.scrollLeft;
  parentRect.right += scrollParent.scrollLeft;
}
```

Since the popper el will have a fixed position, I tell it to get the offsetParent from the reference element instead. All poppers in my application now work correctly (both fixed and non). Of course, that alone is not a great test of the effects of this change...

Should I make a pull request?

**Comment by piehouserat** [comment]
Just thought I would say that I've had success implementing your changes @hgascoigne so thanks for that. I haven't done any extensive testing just yet but I was having issues with incorrect popper positioning within bootstrap modals and they seem to be fixed now :)

**Comment by FezVrasta** [comment]
I've released 0.5.1 today with this change :)
